webapi/src/main/scala/org/knora/webapi/slice/admin/api/service/GroupRestService.scala

```diff
@@ -61,9 +62,14 @@ final case class GroupRestService(
 
   def putGroup(iri: GroupIri, request: GroupUpdateRequest, user: User): Task[GroupGetResponseADM] =
     for {
+      _ <- ZIO
+             .fail(BadRequestException("No data would be changed. Aborting update request."))
+             .when(List(request.name, request.descriptions, request.status, request.selfjoin).flatten.isEmpty)
+      groupToUpdate <- groupService
+                         .findById(iri)
+                         .someOrFail(NotFoundException(s"Group <${iri.value}> not found."))
       _        <- auth.ensureSystemAdminOrProjectAdminOfGroup(user, iri)
```

Comment:
**suggestion**: I think the authorisation check should be done first.

Reply:
When it is like that, in the test `return 'NotFound' if a not-existing group IRI is submitted during update` I'm getting `ForbiddenException` instead of expected `NotFoundException`. Not sure why...

Reply:
You could provide a user in the test which is actually allowed.